### ISSUE TITLE
[Bug] echarts import map bug(empty canvas, but hover can see tips)

### ISSUE THREAD

**Issue by CHENJIAMIAN** [apache/echarts#19414]
### Version

5.4.3

### Link to Minimal Reproduction

https://codepen.io/571574085/pen/rNPXPZg

### Steps to Reproduce

Open It, Wait Loaded. Hover Charts.

### Current Behavior

https://codepen.io/571574085/pen/rNPXPZg

![image](https://github.com/apache/echarts/assets/20126997/2444b9d9-3eba-40f6-9254-a656a8fd45a5)

### Expected Behavior

empty

**Comment by helgasoft** [comment]
cannot reproduce, works as expected - [Demo Code](https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWsDGwA2wBOAXLANoDkAxABwAMAYrQIICspANLBddQCLf1sdyAZgDsDAEz92Fel0oiBM2gCFaXUgF1WaWJDDYApkVTp0YAwA8wRUgHE8AQwAmEA9DCwAymAcYA1gZOsAx4Bg6wAMIAFg54YKQ6AL7aZsA4kCDGOmZ4EADmeQaEHA4WEADObNmwpRUACsAQ7kVZpqZgAJ4gRhwYeMDllSltsNgOAEYG2K0j6OO-fnn9AK7QThE4-DbkAGwOIgCclCzV6InV52fDhoVrM-hODj5EZAAyTQawAIyK79Cf4l-H1gQiB_1gABYwZ8WBoksMwGlsONgBZ7rAAGZhMDLULo9DlBwANwMDHKAEkALYOQrGS6menJHRLCBOdGGDHWDhCACkVTauTyUS5pAhfOGc2AYERlJsvP5piw7gcTVeEymRDAeGWBnhOgsDDK5Re1RMI063RsGCeBjy-A6CraKNWjzwHVsDkymIc2HKBglpkezxIpAAsjBFAAVHWKADqgSjUWWilouUUnie6dWmguOi0Og6hoqJraZraFp6pCJPpjufQ-YJRVcxpIptOsGgDkplb-nx-AbMXUr2A-jtM5R8_hskalPrHBMpaTAUU12v97ZH_28HUM-NMAHdWcuiNR20zZuUosB954OpSUdNvb717NYmFt7vkO30KBfBBOieAB0lADoqmzFP8-6wAYGAxHE5SAUsnpRBAGCAb2sT2M4rjuAAFNQ7AEbARFfOwxDfqYZazKYwAYhifpcgRFFtFguDFKQeB5OMuFfOIIGwOITBMOwXw7EwACUCTUQyoGlsxNF0QxRCkfJ6CsVsHCcdxpHfAcOniEIOySfJ9KzBo4lnqBBiUiAMTlMWX7SRiwAYMsLakH6uQGJUlntkGDgvF8EJEQZ4gidQ-k7BC7AHERQjBewgnUHCbTnpR7adt2Ni9gJ866EONibgYeUTgs06ztgJWLlKK66GuoFFR-PRUW0h5OMexG-ReV43neD5EBiPp-qBb4OE1e4_iAf4AcRwGyZg4FEJB0GwbEYAIUhtmoehHyYY4LhuGA-GESd3xkfJLXUbR9EGIx80sYtmlccdAniPpQnGdJVwXapsDXUpZ2_ep7FabhIgiCJXwHIlH1SV9pkjOZXUjNZtkOPZLaXegzmue5nnNnDqWgf5gXiCFlBhd8Xz6fFiVk-w8VxV8yUXAGl2ZT2wKgqBFaFaOoGlVOHAzj4lUC9VHVajqDUfONjnUW1HWnrMaVtJe163veOADUNL4jKNctY39U0YP-HRASBFHA0tBhQTBcHrYhjhbWhGF4FhB14UR3siedX3y_7_23Se91gWxNig0JIk7JTgkSYT1Gq9RRttEHXIqf7amPRxz3UzsiX6VDn3wxRSMq1ZNl2Q5Rs425Nj495CffbMJMkEI9PfO3iURQztO6SRQghWTKUyW2swc9lwJQjzBUcEVJWTn45Wi1VS61VLettI1nSfkbiu1crIxJwSPWa_1T7De2Bs781FG_qbM3UHNVuPct9trRtzsoa7u3u_tOFHR9sRX2rZ_YpwUjdO6QNs6gzjqdL4Qh44mVDuAn8ilg6A0zgtcOT1tKIPgUwEQxdpIIzaGXI-Fc0YYwmpiFydcOANx8uXPyTwAokHEB3YKschA6QQdFfuiVB4iWHqzMeIwJ4cByiwGelo5783bILJewsKqrxqquaWG5ZY3xofvE8yM1anz6trC-m9TBjEmI-I26t9zqNMW0EAAx_xQGgDYREIAm6wGPjUUIY1tEBxGPfM2Ft5rWw7LbFaDtP7IW2m7D2ACXpAJ0uRMBv004h2gTgnO3E4G6R0tQYhicUGpPQenUOWdMmwPEPwmO7ARBGQ8alUuFlmGzFRlXTGFFa54ybI3fRgZWEvA4XFaglMviUBpvwt6IVYoiSYCzRp9YUCJAANxAA)

**Comment by CHENJIAMIAN** [comment]
Even after removing the code related to Three.js, the charts still don't appear.'

**Comment by helgasoft** [comment]
put all imported .js files in **Settings**, put your code in **JS section**
[Demo](https://codepen.io/helgasoft/pen/gOEppdb)

**Comment by CHENJIAMIAN** [comment]
> put all imported .js files in **Settings**, put your code in **JS section** [Demo](https://codepen.io/helgasoft/pen/gOEppdb) 📌 please close issue if problem solved

Why does this solve the problem? I don't understand the principle behind it. Can you explain it to me?

**Comment by BerndAmend** [comment]
I have the same issue, I'm unable to get the example (https://echarts.apache.org/handbook/en/get-started/#) working if I use esm. (I only see the tooltip)
```import * as echarts from 'https://esm.run/echarts';```
or
```https://cdn.jsdelivr.net/npm/echarts@5.4.3/dist/echarts.js```
It works fine if I use
```<script src="https://cdn.jsdelivr.net/npm/echarts@5.4.3/dist/echarts.js"></script>```

I think something is wrong with the esm files.

**Comment by JimmyAx** [comment]
Can still be reproduced with Apache ECharts 5.5.1.

**Comment by markbaydoun** [comment]
We are having the same bug in our frontend. I've tried downgrading `echarts` to 5.4.1 and `vue-echarts` to 6.6.9, because we recently updated the lockfile in the project and that's the only place where the change could have happened, but the bug still happens. The chart ends up rendering when the page is resized, so I think it's got something to do with how it detects changes and re-renders.

The only way we have been able to "fix" it is by rendering the chart only when a loading state is "false", and show a placeholder otherwise, but this is not ideal, as the loading state of the chart looks much better.

**Comment by vince83110** [comment]
Same error here, with Echarts 6.0.0, and using Symfony project with Importmap component, impossible to have it to work with ESM. Did anyone find a fix ?

For now, I use in my Stimulus JS Controllers:

```
import { Controller } from "@hotwired/stimulus";

export default class extends Controller {
  static targets = ["chart"];

  async connect() {
    const echarts = await import("https://cdn.jsdelivr.net/npm/echarts@5/dist/echarts.esm.min.js");
    const chart = echarts.init(this.chartTarget);
    const option = {
      xAxis: {
        type: 'category',
        data: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun']
      },
      yAxis: {
        type: 'value'
      },
      series: [
        {
          data: [150, 230, 224, 218, 135, 147, 260],
          type: 'line'
        }
      ]
    };

    chart.setOption(option);
  }
}
```

And it works, but not a good solution

**Comment by romulocrj** [comment]
I had the same problem when updating my importmap to echarts version 6 at the URL indicated in jsdelivr: 'https://cdn.jsdelivr.net/npm/echarts@6.0.0/+esm'

However, reverting to the previous version, I realized I wasn't using the URL with the version, but rather the URL: 'https://cdn.jsdelivr.net/npm/echarts/dist/echarts.esm.js'. I believe this points to the latest version of eCharts. But the fact is, the chart works with this URL.

Searching this issue, I discovered that if you use the URL: 'https://cdn.jsdelivr.net/npm/echarts@6/dist/echarts.esm.js' or 'https://cdn.jsdelivr.net/npm/echarts@5/dist/echarts.esm.js' in the importmap, the chart works perfectly.

But I'm not sure if these are the correct versions.

Apparently there is a difference between the esm files in the jsdelivr URLs.

**Comment by wkup** [comment]
### Same symptom under native ESM import maps (no bundler) — root cause + fix
  
  Landed here with the exact symptom (empty `<canvas>`, but tooltips still show on hover) using **ECharts 6.1.0** loaded through a browser **import map** (Symfony AssetMapper) instead of a real bundler. The cause is specific to how the ESM  sub-bundles are served. 
  
  **Setup:** each entry point (`echarts`, `echarts/core`, `echarts/charts`, `echarts/components`, `echarts/renderers`) and every `zrender/lib/*` module was vendored as a **separate jsDelivr-bundled file** and wired individually in the import map. 
  
  **Two failure modes**, both with the model populated (hence working tooltips) but nothing painted on the canvas:
1. Tree-shaken — `import { use } from 'echarts/core'` + `use([LineChart, GridComponent, CanvasRenderer, ...])` throws at `setOption`: Uncaught TypeError: Cannot read properties of undefined (reading 'get') at cartesian2d (charts-*.js) at getInitialData → init → setOption
2. Full import — `import * as echarts from 'echarts'` (the bundled `index.js` that imports `zrender/lib/*` as bare specifiers): **no error at all**, `chart.getOption().series.length === 1`, canvas is allocated at the right size, but `getImageData()` reports **0 painted pixels**.
  
  **Why:** when the sub-modules load as **distinct ESM module instances**, ECharts' internal singleton registries — notably the coordinate-system registry  that `cartesian2d` reads via `CoordinateSystem.get(...)` — are **not shared**   across them. `GridComponent` registers `cartesian2d` into one copy of the   registry; `LineChart.getInitialData()` reads from another → `undefined`. In the   full-import variant the split happens on `zrender`, so the painter is wired to a   different instance and silently draws nothing.
  
  **Fix:** serve a single self-contained bundle, so there is exactly one module  instance → one registry → one painter. The prebuilt full dist ESM bundle inlines both `zrender` and `tslib` (no external/bare imports): echarts/dist/echarts.esm.min.js   // 6.1.0, ~1.1 MB
  
  Symfony AssetMapper `importmap.php`:
  ```php
  // before (broken): split entries — index.js pulls zrender/* as separate modules
  'echarts'            => ['version' => '6.1.0'],
  'echarts/core'       => ['version' => '6.1.0'],
  'echarts/charts'     => ['version' => '6.1.0'],
  'echarts/components' => ['version' => '6.1.0'],
  'echarts/renderers'  => ['version' => '6.1.0'],
  'zrender/lib/...'    => ['version' => '6.1.0'], // ~45 modules

  // after (works): one self-contained file; app code keeps `import * as echarts from 'echarts'`
  'echarts' => ['path' => './assets/lib/echarts.esm.min.js'],

  Charts render normally afterwards (non-blank canvas, no console error).

  TL;DR for anyone on a no-bundler / import-map / SystemJS setup: if you wire the per-module ESM files separately and get an empty canvas (tooltips still work) or cartesian2d … reading 'get', switch to the single dist/echarts.esm.min.js bundle so all internal registries live in one module instance.

**Comment by noproblemwiththat** [comment]
Issue persists with 6.1.0, I confirm that replacing 
```
https://cdn.jsdelivr.net/npm/echarts@6.1.0/+esm
```
by
```
https://cdn.jsdelivr.net/npm/echarts@6.1.0/dist/echarts.esm.min.js
```
fixes it.
